### PR TITLE
[WIP] signing & validation of txs

### LINKED ISSUE #185: Peer suspicion is needed
Given 4 servers in the following validation chain:

**A** --- **B** --- **C** --- **D**

{**A**,**B**,**C**} will be the validators (with **C** as the *proxy tail*).

If an error occurs in validation it is important to be able to assert blame, for example, if **B** receives a transaction and then doesn't validate it, then **B** has a fault and should be moved to the end of the validation chain.

This can be accomplished by signing transactions when they are received, sending cryptographic proof of receipt back to the sender. 

For example, if **A** sends a transaction to **B**, then **B** signs the hash of the transaction with its private key, sending just the signature back to **A**. If **B** never sends the transaction to **C** or lies and says that **A** never sent it, **A** can use the signed receipt from **B** to prove to the other peers that **B** is the faulty server and then they can execute a view change that reorders the peers.

### PR DESCRIPTION
**Pull request by l4l**
resolves #185

### CODE REVIEW

File: core/validation/transaction_validator.cpp [hunk 1]
```diff
@@ -13,21 +13,39 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+    }
 
     template<>
-    bool isValid<ConsensusEvent>(
-        const std::unique_ptr<ConsensusEvent>& tx
-    ){
-        // Write domain logic
-        return true;
+    bool signaturesAreValid<ConsensusEvent>(const std::unique_ptr<ConsensusEvent>& ev) {
+        auto s = ev->eventsignatures();
```
Comment (Warchant) [inline core/validation/transaction_validator.cpp]: Maybe `const auto& s`?

File: core/validation/transaction_validator.cpp [hunk 2]
```diff
@@ -13,21 +13,39 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+        auto tx = ev->transaction();
```
Comment (Warchant) [inline core/validation/transaction_validator.cpp]: the same

File: core/validation/transaction_validator.cpp [hunk 3]
```diff
@@ -13,21 +13,39 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+        return isValid(s, tx.hash());
     }
 
+    template<>
+    bool signaturesAreValid<Transaction>(const std::unique_ptr<Transaction>& tx) {
+        auto s = tx->txsignatures();
```
Comment (Warchant) [inline core/validation/transaction_validator.cpp]: the same as above

File: core/validation/transaction_validator.cpp (hunk 1 above)
Comment (l4l) [inline core/validation/transaction_validator.cpp]: That implies from fn signature

**Pull request merged**